jms/org.wso2.carbon.transport.jms/src/main/java/org/wso2/carbon/transport/jms/listener/JMSServerConnector.java

```diff
@@ -0,0 +1,170 @@
+    private CarbonMessageProcessor carbonMessageProcessor;
+    private JMSConnectionFactory jmsConnectionFactory = null;
+    private Connection connection;
+    private Session session;
+    private Destination destination;
+    private MessageConsumer messageConsumer;
+    private Properties properties;
+    private long retryInterval = 10000;
+    private int maxRetryCount = 5;
+
+    public JMSServerConnector(String id) {
+        super(id);
+    }
+    public JMSServerConnector() {
+        super("jms");
+    }
+
+    void createMessageListener() throws JMSConnectorException {
+        try {
+            connection = jmsConnectionFactory.createConnection();
+            connection.setExceptionListener(new JMSExceptionListener(this, retryInterval, maxRetryCount));
+            jmsConnectionFactory.start(connection);
+            session = jmsConnectionFactory.createSession(connection);
+            destination = jmsConnectionFactory.getDestination(session);
+            messageConsumer = jmsConnectionFactory.createMessageConsumer(session, destination);
+            messageConsumer.setMessageListener(
+                    new JMSMessageListener(carbonMessageProcessor, id, session.getAcknowledgeMode(), session));
+        } catch (RuntimeException e) {
+            logger.error("Error while creating the connection from connection factory. ", e);
+            throw new JMSConnectorException("Error while creating the connection from connection factory", e);
+        } catch (JMSException e) {
+            throw new JMSConnectorException("Error while creating the connection from the connection factory. ", e);
+        }
+    }
+
+    /** Close the connection, session and consumer.
+     * @throws JMSConnectorException Exception that can be thrown when trying to close the connection, session
+     * and message consumer
+     */
+    void closeAll() throws JMSConnectorException {
+        jmsConnectionFactory.closeMessageConsumer(messageConsumer);
+        jmsConnectionFactory.closeSession(session);
+        jmsConnectionFactory.closeConnection(connection);
+    }
+
+    @Override
+    public void setMessageProcessor(CarbonMessageProcessor carbonMessageProcessor) {
+        this.carbonMessageProcessor = carbonMessageProcessor;
+    }
+
+    @Override
+    public void init() throws ServerConnectorException {
+        // not needed
+    }
+
+    @Override
+    public void destroy() throws JMSConnectorException {
+        closeAll();
+    }
+
+
+    @Override
+    public void stop() throws JMSConnectorException {
+       closeAll();
+    }
+
+    @Override
+    protected void beginMaintenance() throws JMSConnectorException {
+        jmsConnectionFactory.stop(connection);
+    }
+
+    @Override
+    protected void endMaintenance() throws JMSConnectorException {
+        jmsConnectionFactory.start(connection);
+    }
+
+    @Override
+    public void start(Map<String, String> map) throws ServerConnectorException {
```

Comment:
shall we inherit docs. Or put a doc comment describing what the input parameter is.